### PR TITLE
Bring markers to front and deselect on empty space click

### LINKED ISSUE #374: Bring selected icon to front
When we have many icons in one area, it would be nice to bring selected one to the front.

Another unrelated thing I was thinking about is to remove selection if you click outside of any icons on the map.

### PR DESCRIPTION
**Pull request by Abyss777**
fix #374 

- combine latest and report layers to one
- bring to front selected marker
- deselect marker (and do other related things) on click empty space

image

image

### CODE REVIEW

File: web/app/view/BaseMap.js [hunk 1]
```diff
@@ -120,9 +120,16 @@ Ext.define('Traccar.view.BaseMap', {
         });
 
         this.map.on('click', function (e) {
-            this.map.forEachFeatureAtPixel(e.pixel, function (feature, layer) {
-                this.fireEvent('selectfeature', feature);
-            }.bind(this));
+            if (this.map.hasFeatureAtPixel(e.pixel, {
+                layerFilter: function (layer) {
+                    return layer.get('name') !== 'geofencesLayer';
+                }})) {
```
Comment (Abyss777) [inline web/app/view/BaseMap.js]: Geofences can be rather big, consider them as empty space.
Also have some doubts in indentation and formatting in this place.

Comment (tananaev) [inline web/app/view/BaseMap.js]: I think you have some problems with formatting here.

File: web/app/view/MapMarkerController.js [hunk 2]
```diff
@@ -346,11 +347,13 @@ Ext.define('Traccar.view.MapMarkerController', {
     selectMarker: function (marker, center) {
         if (this.selectedMarker) {
             this.resizeMarker(this.selectedMarker.getStyle(), false);
+            this.selectedMarker.getStyle().setZIndex(0);
             this.selectedMarker.changed();
         }
 
         if (marker) {
             this.resizeMarker(marker.getStyle(), true);
+            marker.getStyle().setZIndex(1);
```
Comment (tananaev) [inline web/app/view/MapMarkerController.js]: Why is it "1" here? I'm assuming lower index means higher layer, right?

Comment (Abyss777) [inline web/app/view/MapMarkerController.js]: I cant find the documentation explaining this, but if I use -1 marker become lower than others.

**Comment by Abyss777** [comment]
I've changed formatting,  but previous was looking correct, at least `jscs` did not complain about it.

File: web/app/view/BaseMap.js [hunk 3]
```diff
@@ -120,9 +120,15 @@ Ext.define('Traccar.view.BaseMap', {
         });
 
         this.map.on('click', function (e) {
-            this.map.forEachFeatureAtPixel(e.pixel, function (feature, layer) {
-                this.fireEvent('selectfeature', feature);
-            }.bind(this));
+            if (this.map.hasFeatureAtPixel(e.pixel, { layerFilter: function (layer) {
```
Comment (tananaev) [inline web/app/view/BaseMap.js]: It's still not correct. Everything after curly brace should be on a new line. Also there should be no 2 curly braces on the same line at the end. Split it into different lines.

**Comment by Abyss777** [comment]
I've replaced last commit. It's pity that `jscs` can't analyze such complex formatting correctly.

**Comment by tananaev** [comment]
Yes, unfortunately it can't. Thanks for pull request.

**Pull request merged**